pkg/console/subresource/deployment/deployment.go

```diff
@@ -92,116 +107,41 @@ func DefaultDeployment(operatorConfig *operatorv1.Console, cm *corev1.ConfigMap,
+	deployment.Spec.Template.Spec.Containers[0].Command = commands
+	deployment.Spec.Template.Spec.Containers[0].Env = setEnvironmentVariables(proxyConfig)
+	deployment.Spec.Template.Spec.Containers[0].Image = util.GetImageEnv("CONSOLE_IMAGE")
 }
 
 func DefaultDownloadsDeployment(operatorConfig *operatorv1.Console, infrastructureConfig *configv1.Infrastructure) *appsv1.Deployment {
-	labels := util.LabelsForDownloads()
-	meta := util.SharedMeta()
-	meta.Labels = labels
-	meta.Name = api.OpenShiftConsoleDownloadsDeploymentName
-	replicas := Replicas(infrastructureConfig)
-	affinity := downloadsPodAffinity(infrastructureConfig)
-	gracePeriod := int64(0)
-
-	downloadsDeployment := &appsv1.Deployment{
-		ObjectMeta: meta,
-		Spec: appsv1.DeploymentSpec{
-			Replicas: &replicas,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: labels,
-			},
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: api.OpenShiftConsoleDownloadsDeploymentName,
-					Annotations: map[string]string{
-						workloadManagementAnnotation: workloadManagementAnnotationValue,
-					},
-					Labels: labels,
-				},
-				Spec: corev1.PodSpec{
-					NodeSelector: map[string]string{
-						"kubernetes.io/os": "linux",
-					},
-					Affinity:                      affinity,
-					Tolerations:                   tolerations(),
-					PriorityClassName:             "system-cluster-critical",
-					TerminationGracePeriodSeconds: &gracePeriod,
-					Containers: []corev1.Container{
-						{
-							Name:                     "download-server",
-							TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
-							Image:                    util.GetImageEnv("DOWNLOADS_IMAGE"),
-							ImagePullPolicy:          corev1.PullPolicy("IfNotPresent"),
-							Ports: []corev1.ContainerPort{{
-								Name:          api.DownloadsPortName,
-								Protocol:      corev1.ProtocolTCP,
-								ContainerPort: api.DownloadsPort,
-							}},
-							ReadinessProbe: downloadsReadinessProbe(),
-							LivenessProbe:  defaultDownloadsProbe(),
-							Command:        []string{"/bin/sh"},
-							Resources: corev1.ResourceRequirements{
-								Requests: map[corev1.ResourceName]resource.Quantity{
-									corev1.ResourceCPU:    resource.MustParse("10m"),
-									corev1.ResourceMemory: resource.MustParse("50Mi"),
-								},
-							},
-							Args: downloadsContainerArgs(),
-						},
-					},
-				},
-			},
-		},
-	}
+	downloadsDeployment := resourceread.ReadDeploymentV1OrDie(assets.MustAsset("deployments/downloads-deployment.yaml"))
+	withDownloadsReplicas(downloadsDeployment, infrastructureConfig)
+	withDownloadsAffinity(downloadsDeployment, infrastructureConfig)
+	withDownloadsContainers(downloadsDeployment)
 	util.AddOwnerRef(downloadsDeployment, util.OwnerRefFrom(operatorConfig))
 	return downloadsDeployment
 }
 
+func withDownloadsReplicas(downloadsDeployment *appsv1.Deployment, infrastructureConfig *configv1.Infrastructure) {
```

Comment:
do we need a separate function for console (`withReplicas()`) and for downloads (`withDownloadsReplicas()`) ?

Reply:
@jhadvig done!